app/src/androidTest/java/com/android/brewr/ui/e2eTest.kt

```diff
@@ -308,4 +306,48 @@ class E2ETest {
     // Assert the "closed" message is displayed if the list is empty
     composeTestRule.onNodeWithTag("noOpenCoffeeShopsMessage").assertExists()
   }
+
+  @Test
+  fun userProfileScreenE2EFlowTest() {
+    // Step 1: Start from the Overview screen and navigate to the User Profile screen
+    composeTestRule.onNodeWithTag("accountButton").assertIsDisplayed().performClick()
+
+    // Verify User Profile screen is displayed
+    composeTestRule.onNodeWithTag("UserProfileScreen").assertIsDisplayed()
+    composeTestRule.onNodeWithTag("Username").assertIsDisplayed()
+    composeTestRule.onNodeWithTag("User Email").assertIsDisplayed()
+
+    // Step 2: Click on the "Favorite" button to navigate to the User Private List
+    composeTestRule.onNodeWithTag("Favorite button").assertIsDisplayed().performClick()
+
+    // Verify User Private List screen is displayed
+    composeTestRule.onNodeWithTag("UserPrivateListScreen").assertIsDisplayed()
+    composeTestRule.onNodeWithTag("topBar").assertIsDisplayed()
+    composeTestRule.onNodeWithTag("privateList").assertIsDisplayed()
+
+    // Check all elements in User Private List are displayed
+    composeTestRule.onNodeWithTag("coffeeImage:1").performScrollTo().assertIsDisplayed()
+    composeTestRule.onNodeWithTag("coffeeShopName:1").performScrollTo().assertIsDisplayed()
+    composeTestRule.onNodeWithTag("coffeeShopAddress:1").performScrollTo().assertIsDisplayed()
+    composeTestRule.onNodeWithTag("coffeeShopHours:1").performScrollTo().assertIsDisplayed()
+    composeTestRule.onNodeWithTag("coffeeShopRating:1").performScrollTo().assertIsDisplayed()
```

Comment:
We are not sure if coffee will always be present in UserPrivateListScreen with id equal to one, maybe we can try to mock one coffee so that we can always check if there is coffee when testing.